core/control-panel/impl/src/models/registry.rs

```diff
@@ -0,0 +1,79 @@
+use super::artifact::ArtifactId;
+use orbit_essentials::storable;
+use orbit_essentials::types::{Timestamp, UUID};
+use std::collections::BTreeMap;
+
+/// When adding tags to registry entries, if the tag is in this list then it is unique across all the 
+/// entries of the same namespace and name. If the tag exists in other entries, then it will be removed from
+/// older entries.
+/// 
+/// This value is hardcoded, and it should be updated or moved to a configurable value in the future if needed.
+pub const _UNIQUE_TAGS: [&str; 2] = ["latest", "stable"];
+
+/// The registry entry id, which is a UUID.
+pub type RegistryEntryId = UUID;
+
+/// The registry entry is a record that is stored in the registry repository.
+/// 
+/// It stores entries about wasm modules and can be extended to other entry types. When adding new entry types,
+/// the `RegistryEntryValue` enum should be updated to include the new entry type.
+#[storable]
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct RegistryEntry {
+    /// The UUID that identifies the entry in the registry.
+    pub id: RegistryEntryId,
+    /// Wether or not the entry is public, if the entry is public then it is readable by anyone.
+    pub is_public: bool,
+    /// The namespace of the entry, which is used to group entries together (e.g. "@orbit").
+    pub namespace: String,
+    /// The name of the entry, which is used to identify it (e.g. station), within each namespace the
+    /// name should refer to a the same type of entry, but many entries can exist with the same name.
+    ///
+    /// e.g. if the namespace is "@orbit" and the name is "station", then all the entries
+    /// will refer to a wasm module that is a station.
+    pub name: String,
+    /// The tags are used to tag the entry with specific search terms (e.g. "latest", "stable").
+    ///
+    /// Tags are grouped with the format `namespace/name` (e.g. "@orbit/station").
+    pub tags: Vec<String>,
+    /// The category is used to associate the entry with a specific category (e.g. "chain-fusion")
+    /// across all the entries in the registry.
+    pub categories: Vec<String>,
```

Comment:
Should these be part of metadata? (We don't seem to make any programmatic assertions about them anyway.)

Reply:
I think we would add some limits to the number of categories a registry entry can have similar to crates.io, and also, like that would be simpler to create an index with the categories.